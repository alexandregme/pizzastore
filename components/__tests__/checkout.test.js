import React from 'react';
import {shallow} from 'enzyme';
import expect from 'expect';
import Checkout from '../checkout';

describe("Checkout", ()=>{
  let mountedCheckout;

  beforeEach(()=>{
    mountedCheckout = shallow(<Checkout />);
  });

  it('renders without crashing', () => {
    shallow(<Checkout />);
    expect(mountedCheckout).toMatchSnapshot();
  });

  it('expect to have h2 tag', () => {
    let h2 = mountedCheckout.find('h2');
    expect(h2).toHaveLength(1);
    expect(h2.text()).toEqual('Checkout');
  });

  it('expect to have a tag', () => {
    let h2 = mountedCheckout.find('a');
    expect(h2).toHaveLength(1);
  });
});
