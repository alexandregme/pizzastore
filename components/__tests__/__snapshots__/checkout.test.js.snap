// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Checkout renders without crashing 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Checkout />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h2>
          Checkout
        </h2>,
        <p>
          We can't deliver you pizza in the moment but in the mean while you can cook your onw following the instructions bellow.
        </p>,
        <a
          href="http://cooking.nytimes.com/guides/1-how-to-make-pizza"
        >
          http://cooking.nytimes.com/guides/1-how-to-make-pizza
        </a>,
      ],
      "className": "container",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Checkout",
        },
        "ref": null,
        "rendered": "Checkout",
        "type": "h2",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "We can't deliver you pizza in the moment but in the mean while you can cook your onw following the instructions bellow.",
        },
        "ref": null,
        "rendered": "We can't deliver you pizza in the moment but in the mean while you can cook your onw following the instructions bellow.",
        "type": "p",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "http://cooking.nytimes.com/guides/1-how-to-make-pizza",
          "href": "http://cooking.nytimes.com/guides/1-how-to-make-pizza",
        },
        "ref": null,
        "rendered": "http://cooking.nytimes.com/guides/1-how-to-make-pizza",
        "type": "a",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h2>
            Checkout
          </h2>,
          <p>
            We can't deliver you pizza in the moment but in the mean while you can cook your onw following the instructions bellow.
          </p>,
          <a
            href="http://cooking.nytimes.com/guides/1-how-to-make-pizza"
          >
            http://cooking.nytimes.com/guides/1-how-to-make-pizza
          </a>,
        ],
        "className": "container",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Checkout",
          },
          "ref": null,
          "rendered": "Checkout",
          "type": "h2",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "We can't deliver you pizza in the moment but in the mean while you can cook your onw following the instructions bellow.",
          },
          "ref": null,
          "rendered": "We can't deliver you pizza in the moment but in the mean while you can cook your onw following the instructions bellow.",
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "http://cooking.nytimes.com/guides/1-how-to-make-pizza",
            "href": "http://cooking.nytimes.com/guides/1-how-to-make-pizza",
          },
          "ref": null,
          "rendered": "http://cooking.nytimes.com/guides/1-how-to-make-pizza",
          "type": "a",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
